src/main/java/com/spikes2212/command/drivetrains/commands/MoveSmartMotorControllerTankDrivetrain.java

```diff
@@ -0,0 +1,136 @@
+     *
+     * @param subsystem           the {@link SmartMotorControllerGenericSubsystem} this command will run on
+     * @param leftPIDSettings     the left side's loop's PID constants
+     * @param rightPIDSettings    the right side's loop's PID constants
+     * @param feedForwardSettings the loop's feed forward gains
+     * @param controlMode         the loop's control mode (e.g. voltage, velocity, position...)
+     * @param leftSetpoint        the setpoint this command should bring the
+     *                            {@link SmartMotorControllerGenericSubsystem}'s left side to
+     * @param rightSetpoint       the setpoint this command should bring the
+     *                            {@link SmartMotorControllerGenericSubsystem}'s right side to
+     */
+    public MoveSmartMotorControllerTankDrivetrain(SmartMotorControllerTankDrivetrain subsystem,
+                                                  PIDSettings leftPIDSettings, PIDSettings rightPIDSettings,
+                                                  FeedForwardSettings feedForwardSettings,
+                                                  UnifiedControlMode controlMode, Supplier<Double> leftSetpoint,
+                                                  Supplier<Double> rightSetpoint) {
+        addRequirements(subsystem);
+        this.subsystem = subsystem;
+        this.controlMode = controlMode;
+        this.leftPIDSettings = leftPIDSettings;
+        this.rightPIDSettings = rightPIDSettings;
+        this.feedForwardSettings = feedForwardSettings;
+        this.leftSetpoint = leftSetpoint;
+        this.rightSetpoint = rightSetpoint;
+        this.lastTimeLeftNotOnTarget = 0;
+        this.lastTimeRightNotOnTarget = 0;
+    }
+
+    /**
+     * Configures the subsystem's control loops.
+     */
+    @Override
+    public void initialize() {
+        subsystem.configureLoop(leftPIDSettings, rightPIDSettings, feedForwardSettings);
+    }
+
+    /**
+     * Updates any control loops running on the subsystem.
+     */
+    @Override
+    public void execute() {
+        subsystem.pidSet(controlMode, leftSetpoint.get(), rightSetpoint.get(), leftPIDSettings,
+                rightPIDSettings, feedForwardSettings);
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+        subsystem.finish();
+    }
+
+    /**
+     * Checks if the subsystem has been at its target setpoint for longer than its allowed wait time.
+     *
+     * @return {@code true} if the command should finish running, {@code false} otherwise
+     */
+    @Override
+    public boolean isFinished() {
+        if (!subsystem.onTarget(controlMode, leftPIDSettings.getTolerance(), rightPIDSettings.getTolerance(),
+                leftSetpoint.get(), rightSetpoint.get())) {
+            lastTimeLeftNotOnTarget = Timer.getFPGATimestamp();
+            lastTimeRightNotOnTarget = Timer.getFPGATimestamp();
+        }
+        //is this good? it seems wrong
```

Comment:
it's fine but you are calling the `getFPGATimestamp` many times, maybe save this as a variable at the start of the function (`time = Timer.getFPGATimestamp` and use time)